### PR TITLE
libcni: up-convert a Config to a ConfigList when no other configs are found.

### PR DESCRIPTION
**Pull request by squeed**

### CODE REVIEW

**Comment by dcbw** [comment]
LGTM

File: libcni/conf.go [hunk 1]
```diff
@@ -172,7 +181,23 @@ func LoadConfList(dir, name string) (*NetworkConfigList, error) {
...
+
+	// Try and load a network configuration file (instead of list)
+	// from the same name, then upconvert.
+	singleConf, err := LoadConf(dir, name)
+	if err != nil {
+		switch {
+		// no files at all
+		case len(files) == 0 && err == NoConfigsFoundError:
+			return nil, err
+		// some files found
```
Comment (bboreham) [inline libcni/conf.go]: could be clearer - "some files found by `ConfFiles`, but not one matching the required name" ?

File: libcni/conf.go [hunk 2]
```diff
@@ -199,3 +224,29 @@ func InjectConf(original *NetworkConfig, key string, newValue interface{}) (*Net
...
+// ConfListFromConf "upconverts" a network config in to a NetworkConfigList,
+// with the single network as the only entry in the list.
+func ConfListFromConf(original *NetworkConfig) (*NetworkConfigList, error) {
+	// Re-deserialize the config's json, then make a raw map configlist.
+	// This may seem a bit strange, but it's to make the Bytes fields
+	// actually make sense. Otherwise, the generated json is littered with
+	// golang default values.
+
+	rawConfig := make(map[string]interface{})
+	err := json.Unmarshal(original.Bytes, &rawConfig)
```
Comment (bboreham) [inline libcni/conf.go]: could merge with next line

**Comment by tomdee** [comment]
LGTM

**Comment by squeed** [comment]
@bboreham thanks for the feedback, updated.

**Pull request merged**